manager/usermanager/createUser.go

```diff
@@ -26,11 +28,28 @@ func CreateUser(userStore *store.UserStore, user *models.UserCredentials) (*mode
 	}
 
 	password := string(hashedPassword)
-	user.Password = &password
 	uid := uuid.Must(uuid.NewRandom()).String()
-	user.UID = &uid
-	err = userStore.Set(user)
-	return user.GetPublicInfo(), err
+	var role models.Role
+	if user.GetRole() != "" {
```

Comment:
What does this mean?

Reply:
If `userRole` has been passed i.e. it is not empty then set it as the role of user to be created else just keep it as `user`